nvbench/detail/entropy_criterion.cuh

```diff
@@ -33,14 +33,20 @@ class entropy_criterion final : public stopping_criterion_base
   nvbench::int64_t m_total_samples{};
   nvbench::float64_t m_total_cuda_time{};
   std::vector<std::pair<nvbench::float64_t, nvbench::int64_t>> m_freq_tracker;
+  nvbench::float64_t m_sum_count_log_counter{};
 
   // TODO The window size should be user-configurable
   nvbench::detail::ring_buffer<nvbench::float64_t> m_entropy_tracker{299};
 
-  // Used to avoid re-allocating temporary memory
-  std::vector<nvbench::float64_t> m_probabilities;
+  nvbench::float64_t m_sum_x{};
+  nvbench::float64_t m_sum_y{};
+  nvbench::float64_t m_sum_xy{};
+  nvbench::float64_t m_sum_x2{};
+  nvbench::float64_t m_sum_y2{};
```

Comment:
These variables are associated with online computation of linear regression. Instead of introducing them here, could you introduce a class for computing linear regression with online update methods used in this PR?

The class would have `void update( std::pair<double, double> outgoing, std::pair<double, double> incoming)` where pair represents `(x, y)` and `void update( std::pair<double, double> incoming);` methods